src/config.rs

```diff
@@ -24,10 +24,20 @@ impl FromStr for Card {
         if value.len() < 5 {
             return Err(AmdFanError::InputTooShort);
         }
-        value[4..]
+        let card = value[4..]
             .parse::<u32>()
             .map_err(|e| AmdFanError::InvalidSuffix(format!("{:?}", e)))
-            .map(|n| Card(n))
+            .map(|n| Card(n))?;
+        match std::fs::read_to_string(format!("{}/{}/device/vendor", ROOT_DIR, card)) {
```

Comment:
This is really great! I would only ask to change a little bit this part:

```rust
std::fs::read_to_string(format!("{}/{}/device/vendor", ROOT_DIR, card))
             .map_err(|_| AmdFanError::FailedReadVendor)
             .and_then(|vendor| {
                 if vendor.trim() == "0x1002" {
                     Ok(card)
                 } else {
                     Err(AmdFanError::NotAmdCard)
                 }
             })
```

Reply:
Sure thing, lemme change that real quick.